executor/test_linux.h

```diff
@@ -70,7 +70,8 @@ static int test_one(int text_type, const char* text, int text_size, int flags, u
 	if (cpu_mem->exit_reason != reason) {
 		printf("KVM_RUN exit reason %d, expect %d\n", cpu_mem->exit_reason, reason);
 		if (cpu_mem->exit_reason == KVM_EXIT_FAIL_ENTRY)
-			printf("hardware exit reason 0x%llx\n",
+			// __u64 {aka long unsigned int}
+			printf("hardware exit reason 0x%lx\n",
```

Comment:
If you are fixing something, why it did not fail on CI before? Can we make CI detect what you are fixing?

Reply:
Yeah noticed that. On 64bit platform "long int" and "long long int" are the same so it boils down how exactly __u64 is defined on a particular platform. I think I'll keep the format and cast explicitly to (unsigned long long) to shut up gcc.

Reply:
cast it to `uint64`, we specifically define it to work with %llu as far as I remember

Reply:
https://github.com/google/syzkaller/blob/76c77e4180be3d51e960aee84e6d29a36805458a/executor/test_linux.h#L74 is my attempt. This is what "smoke" had to say:
```
gcc -o ./bin/linux_ppc64le/syz-executor executor/executor.cc \
        -D__powerpc64__ -O2 -pthread -Wall -Werror -Wparentheses -Wunused-const-variable -Wframe-larger-than=16384
 -static  -DGOOS_linux=1 -DGOARCH_ppc64le=1 \
        -DHOSTGOOS_linux=1 -DGIT_REVISION=\"82d2e60626ef1f43e557ca2933aee53bd5265eaf+\"
In file included from executor/test.h:5:0,
                 from executor/executor.cc:343:
executor/test_linux.h: In function ‘int test_one(int, const char*, int, int, unsigned int, bool)’:
executor/test_linux.h:74:60: error: format ‘%llx’ expects argument of type ‘long long unsigned int’, but argument
Error: error: format ‘%llx’ expects argument of type ‘long long unsigned int’, but argument
2 has type ‘__u64 {aka long unsigned int}’ [-Werror=format=]
           cpu_mem->fail_entry.hardware_entry_failure_reason);
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
```
Can I please just cast to "unsigned long long" like the rest of that file does?

Reply:
Are you sure this warnings happens with the cast?
The compiler warning output you showed does not contain the cast.
uint64 is already a typedef to long long...